doc/api/async_hooks.md

```diff
@@ -0,0 +1,677 @@
+# Async Hooks
+
+> Stability: 1 - Experimental
+
+The `async-hooks` module provides an API to register callbacks tracking the
+lifetime of asynchronous resources created inside a Node.js application.
+It can be accessed using `require('async-hooks')`.
+
+## Terminology
+
+An async resource represents either a _handle_ or a _request_.
+
+Handles are a reference to a system resource. Some resources are a simple
+identifier. For example, file system handles are represented by a file
+descriptor. Other handles are represented by libuv as a platform abstracted
+struct, e.g. `uv_tcp_t`. Each handle can be continually reused to access and
```

Comment:
Is anything described in the `Some resources … uv_tcp_t` bit actually visible to users? I don’t think so, so I’d just drop it.